rpc/server.go

```diff
@@ -545,6 +603,33 @@ func NewServer(logger utils.ZapCompatibleLogger, opts ...ServerOption) (Server,
 						break
 					}
 					server.mdnsServers = append(server.mdnsServers, mdnsServer)
+
+					// register a second address to match queries for machine-name.local
+					// RSDK-10409 - This may not be necessary
```

Comment:
Maybe this is what you mean by not necessary and (the ticket's description) of too many mdns servers?

But I'd expect this `RegisterProxy` call to happen in the line 574 else-block, but outside of the line 575 "for host in instance names" loop.

Reply:
we do need these changes for pings to machine-name.local to work, and I do believe we need to register an extra mdns server for each instance name (so machine.location.viam + machine.local.location.viam + their dash variants)

the "necessary" part relates to the improvement we talked about around whether we need to register the system hostname as the hostname for the mDNS address (the current state), so if we don't need to do that, we can consider only combining/removing the one of the registrations for each host

Reply:
updated the comments to clarify